src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -44,78 +57,116 @@ class IcebergParquetValueParser {
   static ParquetBufferValue parseColumnValueToParquet(
       Object value,
```

Comment:
For next PR: We need to validate the max nesting depth of the object that's passed in, I think snowflake service caps it at 1000.